PariStyle/src/main/kotlin/domain/usecase/GetClothingRecommendationUseCase.kt

```diff
@@ -1,9 +1,91 @@
 package org.example.domain.usecase
 
 import domain.model.entity.ClothingItem
+import org.example.domain.model.entity.ClothingType
+import org.example.domain.model.entity.TemperatureRange
+import org.example.domain.model.entity.WeatherCondition
+import org.example.domain.model.exception.PariStyleException
 
 class GetClothingRecommendationUseCase(private val getWeatherUseCase: GetWeatherUseCase) {
-    suspend fun getClothingRecommendation(): ClothingItem {
-        TODO("Not yet implemented")
+    suspend fun getClothingRecommendationForCurrentWeather(): ClothingItem {
+        val weatherCondition  = getWeatherUseCase.getWeather().weatherCondition
+        val weatherTemperature = getWeatherUseCase.getWeather().temperature
+        return getClothesItem(weatherCondition, weatherTemperature)
+    }
+
+
+    private fun getClothesItem(weatherCondition: WeatherCondition, temperature : Double): ClothingItem{
+        return pickSuitableClothes(weatherCondition,temperature)
+    }
+
+    private fun pickSuitableClothes(weatherCondition: WeatherCondition, temperature : Double): ClothingItem{
+       val suitableClothes =  getListOfClothesItemsWithDifferentConditions().filter { clothingItem ->
+            clothingItem.suitableConditions.contains(weatherCondition) &&
+                    checkTemperatureRange(temperature,clothingItem)
+        }
+        return suitableClothes.firstOrNull() ?: throw PariStyleException.SuitableClothesException(
+            "No Suitable Clothes available"
+        )
+    }
+
+    private fun getListOfClothesItemsWithDifferentConditions(): List<ClothingItem>{
```

Comment:
we can rename this to availableClothingItems and change it to val
```suggestion
        private val availableClothingItems = listOf(
```

Reply:
maybe I should add UNKNOWN word to the enum class related to weatherCondition so I can use it in an object as empty clothingItem. what do you think ?

Reply:
I think yes it would be good

Reply:
and the variable, what about htis : clothingItemOptions instead of availableClothingItem ? , as they are options an possibilities I created so I feel it fits more